ServiceLayer/Services/CourseService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using DatabaseStructure;
+using DatabaseStructure.Models;
+using ServiceLayer.ServiceInterfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ServiceLayer.Services
+{
+    public class CourseService : ICourseService
+    {
+        private readonly DBContext context;
+
+        public CourseService(DBContext context)
+        {
+            this.context = context;
+        }
+
+        public void AddCourseInDb(string name, int uniquecode, bool isActive)
+        {
+            var course = new Course()
+            {
+                Name = name,
+                UniqueCode = uniquecode,
+                IsActive = isActive
+            };
+            context.Courses.Add(course);
+            context.SaveChanges();
+        }
+
+        public void AddProfessor(int professorId, int courseId)
+        {
+            var course = context.Courses.FirstOrDefault(c => c.UniqueCode == courseId);
+            var professor = context.Professors.FirstOrDefault(p => p.ProfessorId == professorId);
+            course.CourseProfessor = professor;
```

Comment:
After adding FK in the first PR, you can just use something like that: ```course.ProfessorId = professorId``` and remove row 36 to avoid the extra call of DB. Please check for other methods of the other services